### PR TITLE
FUSETOOLS-3245 - ensuring that we always insert a new rest

### PR DESCRIPTION
**Pull request by bfitzpat**
element it goes after the Rest Configuration node.

- tested in the case with no route, elements added correctly
- tested in the case with an existing route, elements added correctly

Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
Before this change, it only worked in the case where an element existed in the context. Now it works in that case as well as the case where no route exists.

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java [hunk 1]
```diff
@@ -518,7 +518,16 @@ public void addRestElement(AbstractCamelModelElement def) {
 		} else {
 			// if it's already there, it's probably in the wrong place - have to move it
 			getXmlNode().removeChild(def.getXmlNode());
-			getXmlNode().insertBefore(def.getXmlNode(), getRestElementEntryPoint());
+			Node entryPoint = getRestElementEntryPoint();
+			if (entryPoint != null) {
+				Element entryPointElement = (Element) entryPoint;
+				if (REST_CONFIGURATION_NODE_NAME.equals(entryPointElement.getTagName())) {
```
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java]: it needs to be compared to (I think) the localName, not the tagname.
it is working with the tagname only if there is no namespace prefix available.
I tried to change to getLocalName but it is working only if we saved the file, it seems the node is not correctly initialized when w have just created it.

the use case, start with
```
<?xml version="1.0" encoding="UTF-8"?>
<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="              http://www.osgi.org/xmlns/blueprint/v1.0.0 https://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd              http://camel.apache.org/schema/blueprint http://camel.apache.org/schema/blueprint/camel-blueprint.xsd">
    <bean class="org.apache.aries.simple.Account" id="accountOne"/>
    <bean class="org.apache.aries.simple.Account2" id="accountTwo">
        <argument value="1"/>
    </bean>
    <bean class="org.apache.aries.simple.Account3" id="accountThree">
        <argument value="1"/>
        <property name="description" value="#1 account"/>
    </bean>
    <c:camelContext id="cbr-example-context" xmlns:c="http://camel.apache.org/schema/blueprint">
        <c:restConfiguration bindingMode="off" host="localhost"/>
    </c:camelContext>
</blueprint>
```

then go to rest tab and click to add a rest element, it is added at the beginning (and without the namespace prefix but this is another bug that I reported here https://issues.jboss.org/browse/FUSETOOLS-3264 )
```
<?xml version="1.0" encoding="UTF-8"?>
<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="              http://www.osgi.org/xmlns/blueprint/v1.0.0 https://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd              http://camel.apache.org/schema/blueprint http://camel.apache.org/schema/blueprint/camel-blueprint.xsd">
    <bean class="org.apache.aries.simple.Account" id="accountOne"/>
    <bean class="org.apache.aries.simple.Account2" id="accountTwo">
        <argument value="1"/>
    </bean>
    <bean class="org.apache.aries.simple.Account3" id="accountThree">
        <argument value="1"/>
        <property name="description" value="#1 account"/>
    </bean>
    <c:camelContext id="cbr-example-context" xmlns:c="http://camel.apache.org/schema/blueprint">
        <rest id="rest-b0b6c8b0-a0a7-4705-be8b-ef4a08bb3ae7"/>
        <c:restConfiguration bindingMode="off" host="localhost"/>
    </c:camelContext>
</blueprint>
```

Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java]: or use something similar to the getTagNameWithoutPrefix() which seems to workaround the bad initialization

Comment (bfitzpat) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java]: Good suggestion. Made the update, tested locally and all seems well.

Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java]: tested also and it is working well (with still https://issues.jboss.org/browse/FUSETOOLS-3264 but as said before it doesn't come from this PR)

**Pull request merged**